packages/excalidraw/scene/Scene.ts

```diff
@@ -229,12 +235,27 @@ class Scene {
 
   replaceAllElements(
     nextElements: readonly ExcalidrawElement[],
-    mapElementIds = true,
+    reorderedElements?: Map<string, ExcalidrawElement>,
   ) {
-    this.elements = nextElements;
+    let _nextElements;
+    if (reorderedElements) {
+      _nextElements = fixFractionalIndices(nextElements, reorderedElements);
+    } else {
+      _nextElements = nextElements;
+    }
+
+    if (import.meta.env.DEV && import.meta.env.MODE !== ENV.TEST) {
```

Comment:
```suggestion
    if (import.meta.env.DEV && import.meta.env.MODE === ENV.TEST) {
```

Reply:
throwing in test will cause most existing tests to fail, that's why i added the inequality

Reply:
That was the point in https://github.com/excalidraw/excalidraw/pull/7359#discussion_r1431192007

Reply:
@ryan-di let's check whether we can fix all the failing tests, otherwise, this invariant check won't be very helpful.